exampleplugins/5.2/SwagCustomProductBoxLayout/Subscriber/Frontend.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace SwagCustomProductBoxLayout\Subscriber;
+
+use Enlight\Event\SubscriberInterface;
+
+class Frontend implements SubscriberInterface
+{
+    /**
+     * @var string
+     */
+    private $pluginDirectory;
+
+    /**
+     * @param $pluginDirectory
+     */
+    public function __construct($pluginDirectory)
+    {
+        $this->pluginDirectory = $pluginDirectory;
+    }
+    
+    public static function getSubscribedEvents()
+    {
+        return array(
+            'Enlight_Controller_Action_PostDispatchSecure_Frontend_Listing' => 'onFrontendPostDispatchListing',
```

Comment:
if you only use these two events, your template directory will only be loaded on listing pages. this could be a problem, if the product boxes are used in sliders, e.g. on the detail page. did you check that? maybe you should use something like this:
```
    public static function getSubscribedEvents()
    {
        return [
            'Theme_Inheritance_Template_Directories_Collected' => 'onTemplateDirectoriesCollect',
        ];
    }

    public function onTemplateDirectoriesCollect(EventArgs $args)
    {
        $dirs = $args->getReturn();
        $dirs[] = $this->pluginPath . '/Resources/views/';
        $args->setReturn($dirs);
    }
```

Reply:
if you use the suggested event, the addTemplateDir in the backend subscriber should be obsolete, too

Reply:
maybe @Pmuckel could add this event? :see_no_evil: